vcd/resource_vcd_nsxt_alb_pool.go

```diff
@@ -0,0 +1,639 @@
+	if err != nil {
+		return diag.Errorf("error getting NSX-T ALB Pool type: %s", err)
+	}
+	createdAlbPool, err := vcdClient.CreateNsxtAlbPool(albPoolConfig)
+	if err != nil {
+		return diag.Errorf("error setting NSX-T ALB Pool: %s", err)
+	}
+
+	d.SetId(createdAlbPool.NsxtAlbPool.ID)
+
+	return resourceVcdAlbPoolRead(ctx, d, meta)
+}
+
+func resourceVcdAlbPoolUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	albPool, err := vcdClient.GetAlbPoolById(d.Id())
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("could not retrieve NSX-T ALB Pool: %s", err))
+	}
+
+	updatePoolConfig, err := getNsxtAlbPoolType(d)
+	if err != nil {
+		return diag.Errorf("error getting NSX-T ALB Pool type: %s", err)
+	}
+	updatePoolConfig.ID = d.Id()
+
+	_, err = albPool.Update(updatePoolConfig)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error updating NSX-T ALB Pool: %s", err))
+	}
+
+	return resourceVcdAlbPoolRead(ctx, d, meta)
+}
+
+func resourceVcdAlbPoolRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	albPool, err := vcdClient.GetAlbPoolById(d.Id())
+	if err != nil {
+		if govcd.ContainsNotFound(err) {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(fmt.Errorf("could not retrieve NSX-T ALB Pool: %s", err))
+	}
+
+	err = setNsxtAlbPoolData(d, albPool.NsxtAlbPool)
+	if err != nil {
+		return diag.Errorf("error setting NSX-T ALB Pool data: %s", err)
+	}
+	d.SetId(albPool.NsxtAlbPool.ID)
+	return nil
+}
+
+func resourceVcdAlbPoolDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	albPool, err := vcdClient.GetAlbPoolById(d.Id())
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("could not retrieve NSX-T ALB Pool: %s", err))
+	}
+
+	err = albPool.Delete()
+	if err != nil {
+		return diag.Errorf("error deleting NSX-T ALB Pool: %s", err)
+	}
+
+	return nil
+}
+
+func resourceVcdAlbPoolImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[TRACE] NSX-T ALB Pool import initiated")
+
+	resourceURI := strings.Split(d.Id(), ImportSeparator)
+	if len(resourceURI) != 4 {
+		return nil, fmt.Errorf("resource name must be specified as org-name.vdc-name.nsxt-edge-gw-name.pool_name")
+	}
+	orgName, vdcName, edgeName, poolName := resourceURI[0], resourceURI[1], resourceURI[2], resourceURI[3]
+
+	vcdClient := meta.(*VCDClient)
+
+	_, vdc, err := vcdClient.GetOrgAndVdc(orgName, vdcName)
+	if err != nil {
+		return nil, fmt.Errorf("unable to find VDC %s: %s", vdcName, err)
+	}
+
+	if vdc.IsNsxv() {
+		return nil, fmt.Errorf("ALB Pools are only supported on NSX-T please use 'vcd_lb_server_pool' for NSX-V load balancers")
```

Comment:
If you're capitalizing "Pools" then probably "Load Balancers" should be as well?
```suggestion
		return nil, fmt.Errorf("ALB Pools are only supported on NSX-T please use 'vcd_lb_server_pool' for NSX-V Load Balancers")
```

Reply:
Changed and this and a few more capitalization cases